go/chat/signencrypt/codec_test.go

```diff
@@ -0,0 +1,550 @@
+package signencrypt
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"golang.org/x/crypto/nacl/secretbox"
+
+	"github.com/agl/ed25519"
+	"github.com/stretchr/testify/require"
+)
+
+var plaintextInputs = []string{
+	"",
+	"1",
+	"short",
+	strings.Repeat("long", 1000000),
+}
+
+func zeroSecretboxKey() SecretboxKey {
+	var key [SecretboxKeySize]byte // all zeroes
+	return &key
+}
+
+func zeroNonce() Nonce {
+	var nonce [NonceSize]byte // all zeroes
+	return &nonce
+}
+
+func zeroVerifyKey() VerifyKey {
+	var key [ed25519.PublicKeySize]byte
+	// Generated from libsodium's crypto_sign_seed_keypair with a zero seed.
+	copy(key[:], ";j'\xbc\xce\xb6\xa4-b\xa3\xa8\xd0*o\rse2\x15w\x1d\xe2C\xa6:\xc0H\xa1\x8bY\xda)")
```

Comment:
It turns out that `\x##` in a Go string means "literal byte" and does not refer to unicode code points. So every character/hex-thing here is being interpreted as a single byte, and the string itself is not valid UTF8 at all. (This is different from what Python does, where `\xff` would be the _code point_ 0xFF and would get encoded as two bytes in UTF8.)